### DISCUSSION TITLE
Hyperelasticity - can we do it in MOOSE?

### DISCUSSION THREAD

**Discussion by WilkAndy** [idaholab/moose#24637] [Q&A Modules: Solid mechanics] [answered]
Hey folks,

My specific question is I'm wondering whether TensorMechanics can model Yeoh hyperelasticity, where the energy density is

$$\sum_{n=1}^{3} A_n (I_1 - 3)^n $$

with $A_n$ specified by the user, and $I_1$ is the first invariant of the Cauchy-Green deformation tensor.

More generally, and probably less usefully, can we model nonlinear elasticity?  Eg, i could differentiate the above with respect to strain to get stress = nonlinear function of strain, and then insert the correct coefficients (=material moduli) into a nonlinear elasticity Material class, if TensorMechanics has one.

**Comment by hugary1995** [comment, marked as answer]
ya we have 
- ComputeLagrangianStressPK1
- ComputeLagrangianStressPK2
- ComputeLagrangianStressCauchy

Depending on your stress measure.
